packages/react-server/src/ReactFizzServer.js

```diff
@@ -1132,9 +1176,37 @@ function validateIterable(iterable, iteratorFn: Function): void {
   }
 }
 
+let renderNodeDestructive = renderNodeDestructiveImpl;
```

Comment:
Don't use mutations in module scope. This messes with the compiler optimizations and resolving of cyclic dependencies because now things start getting dependent on ordering.

Instead create a function that inside of it does the DEV check.